contracts/modules/capital/SwapOperator.sol

```diff
@@ -86,129 +78,270 @@ contract SwapOperator {
 
   receive() external payable {}
 
-  /**
-   * @dev Compute the digest of an order using CoW protocol's logic
-   * @param order The order
-   * @return The digest
-   */
+  /// @dev Compute the digest of an order using CoW protocol's logic
+  /// @param order The order
+  /// @return The digest
   function getDigest(GPv2Order.Data calldata order) public view returns (bytes32) {
     bytes32 hash = GPv2Order.hash(order, domainSeparator);
     return hash;
   }
 
-  /**
-   * @dev Compute the UID of an order using CoW protocol's logic
-   * @param order The order
-   * @return The UID (56 bytes)
-   */
+  /// @dev Compute the UID of an order using CoW protocol's logic
+  /// @param order The order
+  /// @return The UID (56 bytes)
   function getUID(GPv2Order.Data calldata order) public view returns (bytes memory) {
     bytes memory uid = new bytes(56);
     bytes32 digest = getDigest(order);
     GPv2Order.packOrderUidParams(uid, digest, order.receiver, order.validTo);
     return uid;
   }
+  
+  /// @dev Using oracle prices, returns the equivalent amount in `toAsset` for a given `fromAssetAmount` in `fromAsset`
+  /// Supports conversions for ETH to Asset, Asset to ETH, and Asset to Asset
+  function getOracleAmount(address toAsset, address fromAsset, uint fromAssetAmount) internal view returns (uint) {
+    IPriceFeedOracle priceFeedOracle = _pool().priceFeedOracle();
+
+    if (fromAsset == address(weth)) {
+      // ETH -> toAsset
+      return priceFeedOracle.getAssetForEth(toAsset, fromAssetAmount);
+    }
+    if (toAsset == address(weth)) {
+      // fromAsset -> ETH
+      return priceFeedOracle.getEthForAsset(fromAsset, fromAssetAmount);
+    }
+    // fromAsset -> toAsset via ETH
+    uint fromAssetInEth = priceFeedOracle.getEthForAsset(fromAsset, fromAssetAmount);
+    return priceFeedOracle.getAssetForEth(toAsset, fromAssetInEth);
```

Comment:
The oracle may always have a slight error because it's normally lagging (due to max deviation % and heartbeat). Given we're using two oracles here the error might compound - something to keep in mind when setting the max slippage on assets.